tools/ack-discover/controller.py

```diff
@@ -39,36 +47,50 @@ class Controller:
     source_repo_url: str = None
     image_repo: str = None
     chart_repo: str = None
+    gh_issue_url: str = None
 
 
 def collect_all(writer, gh, ep_client, services):
     """Returns a map, keyed by service package name, of ControllerInfo objects
     describing the ACK controllers.
     """
-    writer.debug("[collect_controllers] collecting ACK controller information ... ")
+    writer.debug("[controller.collect_all] collecting ACK controller information ... ")
     ack_org = gh.get_organization("aws-controllers-k8s")
     result= {}
 
     for service_package_name, service in services.items():
-        writer.debug(f"[collect_controllers] finding controller info for {service_package_name} ...")
+        writer.debug(f"[controller.collect_all] finding controller info for {service_package_name} ...")
+
+        project_stage = project_stages.NONE
+        maintenance_phase = maintenance_phases.NONE
+        # We check if there has been a GH issue created for the AWS service and
+        # if that GH issue has been placed in the Service Controller Github
+        # Project's "Planned" ProjectColumn.
+        gh_issue = get_controller_request_issue(writer, gh, service)
+        gh_issue_url = None
+        if gh_issue is not None:
+            gh_issue_url = gh_issue.html_url
+            project_stage = project_stages.PROPOSED
+            if is_planned(writer, gh, gh_issue):
+                project_stage = project_stages.PLANNED
+
         try:
             repo = ack_org.get_repo(service_package_name+"-controller")
         except github.UnknownObjectException:
             controller = Controller(
                 service=service,
                 latest_release=None,
-                project_stage=project_stages.NONE,
-                maintenance_phase=maintenance_phases.NONE,
+                project_stage=project_stage,
+                maintenance_phase=maintenance_phase,
                 source_repo_url=None,
                 image_repo=None,
                 chart_repo=None,
+                gh_issue_url=gh_issue_url,
             )
             result[service_package_name] = controller
```

Comment:
good question... in the case of EC2, we already have a controller so the service record's project phase is set to RELEASED. But for other services where we have multiple GH issues we will need to manually handle those I think.

Reply:
> But for other services where we have multiple GH issues we will need to manually handle those I think.

I don't quite understand why services have multiple GH issues labeled `Service Controller` if the intent is each of these map to a single repo (or a single service API in `aws-sdk`).  Ex EC2:
  * https://github.com/aws-controllers-k8s/community/issues/489
  * https://github.com/aws-controllers-k8s/community/issues/490

**[Brainstorm]** This code could be a good opportunity to "enforce" GH issue structure in that it expects 1 repo per `Service Controller` GH issue. We can consolidate existing issues today and since "services" is a known list with _infrequent_ updates we can create all GH issues ahead of time for each service aligning with the expected format.

Reply:
@brycahta agreed, we could definitely take the opportunity to clean up our GH issues and standardize the service controller release roadmap project. For this particular PR, though, it's solving the initial problem at hand, which is to properly list PROPOSED and PLANNED project stages for specific services. This is important because I'm trying to replace the hand-edited services matrix with a table that is generated from the data produced by the `ack-discover` tool...